pilot/pkg/model/context.go

```diff
@@ -643,14 +643,14 @@ func (node *Proxy) GetNetworkView() map[string]bool {
 
 // InNetwork returns true if the proxy is on the given network, or if either
 // the proxy's network or the given network is unspecified ("").
-func (node *Proxy) InNetwork(network string) bool {
-	return node == nil || SameOrEmpty(network, node.Metadata.Network)
+func (node *Proxy) InNetwork(network NetworkID) bool {
+	return node == nil || SameOrEmpty(string(network), string(node.Metadata.Network))
```

Comment:
Consider defining a String() for the type NetworkID to have a better control of how the type NetworkID is converted to string.